sqlglot/dialects/bigquery.py

```diff
@@ -401,6 +401,9 @@ class Parser(parser.Parser):
             ),
             "TIMESTAMP_SECONDS": lambda args: exp.UnixToTime(this=seq_get(args, 0)),
             "TO_JSON_STRING": exp.JSONFormat.from_arg_list,
+            "FORMAT_DATETIME": lambda args: exp.TimeToStr(
+                this=exp.TsOrDsToTimestamp(this=seq_get(args, 1)), format=seq_get(args, 0)
+            ),
```

Comment:
It looks like this essentially maps `FORMAT_DATETIME` to `FORMAT_DATE`. Is this conversion safe? Can you share docs / examples that indicate these two are equivalent? Make sure to test various types for the input arguments, because there may be subtle differences there.

Reply:
With the CAST to DATETIME, the FORMAT_DATETIME and FORMAT_DATE functions work the same for TIMESTAMP, DATETIME, and DATE arguments.
Both functions accept the same format elements (Doc: https://cloud.google.com/bigquery/docs/reference/standard-sql/format-elements#format_elements_date_time).
This query returns TRUE for all comparisons, demonstrating the equivalence of these functions when CAST as DATETIME is used in the FORMAT_DATE function.

SELECT 
    -- Example 1: Full weekday name (DATETIME)
    FORMAT_DATETIME('%A', CAST('2024-08-11 14:30:45' AS DATETIME)) = FORMAT_DATE('%A', CAST('2024-08-11 14:30:45' AS DATETIME)) AS datetime_full_weekday_name,

    -- Example 2: Abbreviated weekday name (DATETIME)
    FORMAT_DATETIME('%a', CAST('2024-08-11 14:30:45' AS DATETIME)) = FORMAT_DATE('%a', CAST('2024-08-11 14:30:45' AS DATETIME)) AS datetime_abbreviated_weekday_name,

    -- Example 3: Full month name (TIMESTAMP)
    FORMAT_DATETIME('%B', CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP)) = FORMAT_DATE('%B', CAST(CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP) AS DATETIME)) AS timestamp_full_month_name,

    -- Example 4: Abbreviated month name (TIMESTAMP)
    FORMAT_DATETIME('%b', CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP)) = FORMAT_DATE('%b', CAST(CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP) AS DATETIME)) AS timestamp_abbreviated_month_name,

    -- Example 5: Day of the month (DATETIME)
    FORMAT_DATETIME('%d', CAST('2024-08-11 14:30:45' AS DATETIME)) = FORMAT_DATE('%d', CAST('2024-08-11 14:30:45' AS DATETIME)) AS datetime_day_of_month,

    -- Example 6: Day of the year (TIMESTAMP)
    FORMAT_DATETIME('%j', CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP)) = FORMAT_DATE('%j', CAST(CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP) AS DATETIME)) AS timestamp_day_of_year,

    -- Example 7: ISO 8601 year with century (DATETIME)
    FORMAT_DATETIME('%G', CAST('2024-08-11 14:30:45' AS DATETIME)) = FORMAT_DATE('%G', CAST('2024-08-11 14:30:45' AS DATETIME)) AS datetime_iso_year,

    -- Example 8: ISO 8601 week number (DATE)
    FORMAT_DATETIME('%V', CAST('2024-08-11' AS DATE)) = FORMAT_DATE('%V', CAST(CAST('2024-08-11' AS DATE) AS DATETIME)) AS timestamp_iso_week_number,

    -- Example 9: The week number of the year (Sunday as the first day of the week, DATETIME)
    FORMAT_DATETIME('%U', CAST('2024-08-11 14:30:45' AS DATETIME)) = FORMAT_DATE('%U', CAST('2024-08-11 14:30:45' AS DATETIME)) AS datetime_week_number_sunday_start,

    -- Example 10: Minute as a decimal number (TIMESTAMP)
    FORMAT_DATETIME('%M', CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP)) = FORMAT_DATE('%M', CAST(CAST('2024-08-11 14:30:45+00:00' AS TIMESTAMP) AS DATETIME)) AS time_minute_format,

    -- Example 11: Second as a decimal number (DATE)
    FORMAT_DATETIME('%S', DATE '2024-08-11') = FORMAT_DATE('%S', CAST(DATE '2024-08-11' AS DATETIME)) AS time_second_format;

-- Note: Neither function accepts TIME arguments, as shown below.
SELECT FORMAT_DATETIME('%M', TIME '2024-08-11 14:30:45') AS time_minute_format;  -- This will cause an error.
SELECT FORMAT_DATE('%H', TIME '2024-08-11 14:30:45') AS time_minute_format;      -- This will cause an error.